### ISSUE TITLE
[rustdoc] Misleading error message for ambiguous link to associated type/const

### ISSUE THREAD

**Issue by ModProg** [rust-lang/rust#108653]
### Code

```Rust
/// [`Trait::IDENT`]
trait Trait {
    type IDENT;
    const IDENT: usize;
}
```

### Current output

```Shell
warning: `Trait::IDENT` is both a trait and a trait
 --> src/main.rs:1:7
  |
1 | /// [`Trait::IDENT`]
  |       ^^^^^^^^^^^^ ambiguous link
  |
  = note: `#[warn(rustdoc::broken_intra_doc_links)]` on by default
help: to link to the trait, prefix with `trait@`
  |
1 | /// [`trait@Trait::IDENT`]
  |       ++++++
help: to link to the trait, prefix with `trait@`
  |
1 | /// [`trait@Trait::IDENT`]
  |       ++++++

warning: `doc_` (bin "doc_" doc) generated 1 warning
```

### Desired output

```Shell
warning: `Trait::IDENT` is both a const and a type
 --> src/main.rs:1:7
  |
1 | /// [`Trait::IDENT`]
  |       ^^^^^^^^^^^^ ambiguous link
  |
  = note: `#[warn(rustdoc::broken_intra_doc_links)]` on by default
help: to link to the const, prefix with `const@`
  |
1 | /// [`const@Trait::IDENT`]
  |       ++++++
help: to link to the type, prefix with `type@`
  |
1 | /// [`type@Trait::IDENT`]
  |       ++++++

warning: `doc_` (bin "doc_" doc) generated 1 warning
```

**Comment by ModProg** [comment]
Probably a related error with enum variant + trait:

```rs
enum Enum {
    IDENT,
}

/// [`Self::IDENT`]
impl Trait for Enum {
    type IDENT = usize;

    const IDENT: usize = 10;
}
```

```rs
warning: `Self::IDENT` is both an enum and an enum
  --> src/main.rs:11:7
   |
11 | /// [`Self::IDENT`]
   |       ^^^^^^^^^^^ ambiguous link
   |
help: to link to the enum, prefix with `enum@`
   |
11 | /// [`enum@Self::IDENT`]
   |       +++++
help: to link to the enum, prefix with `enum@`
   |
11 | /// [`enum@Self::IDENT`]
   |       +++++
```
